### PR TITLE
Add quotes around -D option values.  Remove duplicate options

### LINKED ISSUE #26893: Space in value of -D option in jvm.options breaks server package command
**Describe the bug**  
If jvm.options has a -D option where the value contains a space, the server package command does not work.

**Steps to Reproduce**  
For example if jvm.options contains:
-Dpoller.mancapadjimport.time=3:20 PM

command:
./server package myServer --include=minify --archive=/tmp/myServer.zip

Result is:
Error: Could not find or load main class PM 

**Expected behavior**  

expected output
...
Building archive for server myServer.
Server myServer package complete in /tmp/myServer.zip.

**Diagnostic information:**  
 - OpenLiberty Version:  20.0.0.8 - 23.0.0.11
 - Affected feature(s) : server package command
 - Java Version: N/A
 - server.xml:  N/A. default server is fine

### PR DESCRIPTION
**Pull request by jimblye**
fixes #26893

### CODE REVIEW

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server [hunk 1]
```diff
@@ -238,7 +238,7 @@ else
   ##
```
Comment (jimblye) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:238]: This is just a change to the preferred syntax.   Not part of the fix.

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server [hunk 2]
```diff
@@ -238,7 +238,7 @@ else
...
   escapeForEval()
   {
-    escapeForEvalResult=\'`safeEcho "$1" | sed s/\'/\'\"\'\"\'/g`\'
+    escapeForEvalResult=\'$(safeEcho "$1" | sed s/\'/\'\"\'\"\'/g)\'
```
Comment (jimblye) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:241]: preferred syntax only - not part of fix.

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server [hunk 3]
```diff
@@ -916,7 +946,7 @@ readServerEnv()
   if [ -f "$1" ]; then
     saveIFS=$IFS
     IFS=$newline
-    for line in `readNativeFile "$1" '[#_A-Za-z=]' | tr -d '\r'`; do
+    for line in $(readNativeFile "$1" '[#_A-Za-z=]' | tr -d '\r'); do
```
Comment (jimblye) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:920]: again - preferred syntax only

**Review by cbridgha — DISMISSED** [review]
Good with these changes - obviously would like to see several flavors of linux tested etc...

**Comment by jimblye** [comment]
#libby
#build

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server [hunk 4]
```diff
@@ -1011,7 +1012,7 @@ readNativeFile() {
   if ${nativeEBCDIC}; then
     # ASCII 'm' overlaps with EBCDIC '_', so strip it out before detecting the codepage.
     # Note: cat used here to handle ASCII-tagged files.
-    filecontents=`cat "$1" | iconv -f ISO8859-1 -t IBM-1047 | tr -d 'm\r\n'`
+    filecontents=$(cat "$1" | iconv -f ISO8859-1 -t IBM-1047 | tr -d 'm\r\n')
```
Comment (jimblye) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:1015]: preferred syntax

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server [hunk 5]
```diff
@@ -1880,8 +1877,8 @@ case "$ACTION" in
     if checkServer true
     then
       serverPIDStatus
-      JVM_OPTIONS_QUOTED=${SERVER_JVM_OPTIONS_QUOTED}
-      javaCmd '' "${JVM_OPTIONS_QUOTED}" "${SERVER_NAME}" --pid="${PID}" --package "$@" 
+      JVM_OPTIONS_QUOTED=${SERVER_JVM_OPTIONS_QUOTED} 
+      javaCmd '' "${SERVER_NAME}" --pid="${PID}" --package "$@"
```
Comment (jimblye) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:1881]: avoids duplicate options on command line

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server [hunk 6]
```diff
@@ -884,27 +884,28 @@ findRealPath() {
...
-            escapeForEval "${option}"
-            SERVER_JVM_OPTIONS_QUOTED="${SERVER_JVM_OPTIONS_QUOTED} '${option}'"
-            ;;
-          esac
-        fi
-      done
-      IFS=$saveIFS
-    fi
-
+              escapeForEval "${option}"
```
Comment (jimblye) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:900]: escapeForEval doesn't return anything.   The result needs to be assigned separately.  
escapeForEval also surrounds the result in quotes.  So adding another set of quotes around the result becomes unnecessary.

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server [hunk 7]
```diff
@@ -884,27 +884,28 @@ findRealPath() {
...
+              option="${escapeForEvalResult}"
+              SERVER_JVM_OPTIONS_QUOTED="${SERVER_JVM_OPTIONS_QUOTED} ${option}"
+              ;;
```
Comment (jimblye) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:903]: escapeForEval doesn't return anything. The result needs to be assigned separately.
escapeForEval also surrounds the result in quotes. So adding another set of quotes around the result becomes unnecessary.

**Comment by jimblye** [comment]
#build
#libby

**Review by tbitonti — APPROVED** [review]

**Comment by jimblye** [comment]
#libby

**Pull request merged**